src/main/java/org/example/jvspringbootfirstbook/service/order/OrderServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+        order.setShippingAddress(requestDto.shippingAddress());
+        order.setOrderItems(new HashSet<>());
+        return orderRepository.save(order);
+    }
+
+    private Set<OrderItem> calculatePriceForSet(Set<OrderItem> orderItemSet) {
+        return orderItemSet.stream()
+                .map(this::setPrice)
+                .collect(Collectors.toSet());
+    }
+
+    private OrderItem setPrice(OrderItem orderItem) {
+        orderItem.setPrice(
+                orderItem.getBook().getPrice().multiply(
+                        BigDecimal.valueOf(orderItem.getQuantity())
+                )
+        );
+        return orderItem;
+    }
+
+    private Set<OrderItem> getOrderItemSet(ShoppingCart cart, Order emptyOrder) {
+        Set<CartItem> cartItems = cart.getCartItems();
+        return cartItems.stream()
+                .map(orderItemMapper::fromCarttoOrderItem)
+                .map(orderItem -> setOrderToItem(orderItem, emptyOrder))
+                .collect(Collectors.toSet());
+    }
+
+    private OrderItem setOrderToItem(OrderItem orderItem, Order emptyOrder) {
+        orderItem.setOrder(emptyOrder);
+        return orderItem;
+    }
+
+    @Override
+    public List<OrderDto> getHistory(User user, Pageable pageable) {
+        return orderRepository.findAllByUser(user, pageable).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderItemDto> getFromOrder(Long orderId) {
+        Order order = getOrderFromId(orderId);
+        return order.getOrderItems().stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public OrderItemDto getFromOrder(Long orderId, Long id) {
+        Order order = getOrderFromId(orderId);
+        Set<OrderItem> orderItems = order.getOrderItems();
+        OrderItem orderItem = orderItems.stream()
+                .filter(item -> item.getId().equals(id))
+                .findFirst()
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Order item with id: " + id + " not found"
+                        )
+                );
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    private Order getOrderFromId(Long orderId) {
+        Order order = orderRepository.findById(orderId)
+                .orElseThrow(
+                        () -> new EntityNotFoundException(
+                                "Order with id: " + orderId + " not found")
+                );
+        return order;
+    }
+
+    @Override
+    public OrderDto updateStatus(Long id) {
+        if (!orderRepository.findById(id).isPresent()) {
+            throw new EntityNotFoundException("We don't have a order");
+        }
+        Order order = orderRepository.findById(id)
+                .orElseThrow(
+                        () -> new EntityNotFoundException("We don't have a order")
```

Comment:
As Beata mentioned, you should change the text in these exceptions and since they are all the same, you should make it a constant and use it there:

![image](https://github.com/user-attachments/assets/512c500d-0c95-40e4-b818-a904997216e3)